os/se/ameba/security_ameba_wrapper_tz.c

```diff
@@ -604,9 +609,11 @@ int se_ameba_hal_get_factory_cert(uint32_t cert_idx, hal_data *cert)
 	AWRAP_ENTER;
 	int ret = HAL_SUCCESS;
 
+	device_mutex_lock(RT_DEV_LOCK_FLASH);
```

Comment:
if up_allocate_secure_context uses wait or sleep for allocation, it might create issue in mutex lock?

Reply:
Device mutex lock used sem_timedwait() and sem_post(). From my point of view it will only delay the unlock time.
According to the test results there is no problem.

Reply:
Would it be better if the order was changed? up_allocate_secure_context() follow by device_mutex_lock().